### PR TITLE
group parameter for search URL

### PR DESCRIPTION
**Pull request by tulinkry**
prerequisite for #1974

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java [hunk 1]
```diff
@@ -875,71 +879,90 @@ private static void splitByComma(String value, List<String> result) {
...
+        List<String> names = getParamVals(projectParamName);
+        for (String projectName : names) {
+            Project project = Project.getByName(projectName);
+            if (project != null && authFramework.isAllowed(req, project)) {
+                projectNames.add(projectName);
+            }
+        }
+
+        /**
+         * Add all projects which are part of a group which match the group parameter name
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java]: that matches

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java [hunk 2]
```diff
@@ -794,24 +794,28 @@ public String getRequestedProjectsAsString() {
...
-     * with invalid projects removed gets returned. Otherwise:</li> <li>If a
-     * default project is set in the RTE, this project gets returned.
-     * Otherwise:</li> <li>an empty set</li> </ol>
+     * @return a possible empty set of project names but never {@code null}.
      */
     public SortedSet<String> getRequestedProjects() {
         if (requestedProjects == null) {
             requestedProjects
-                    = getRequestedProjects("project", OPEN_GROK_PROJECT);
+                    = getRequestedProjects("project", "group", OPEN_GROK_PROJECT);
```
Comment (vladak) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java]: define "group" as this final static string ?

Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java]: the same for "project" ? I kept it consisten but I can improve it

**Comment by vladak** [comment]
Generally looks good, kudos for the tests !

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java (hunk 2 above)
Comment (vladak) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java]: yep

Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java]: I did the same for project

File: opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java (hunk 1 above)
Comment (tulinkry) [reply opengrok-indexer/src/main/java/org/opengrok/indexer/web/PageConfig.java]: ok

**Comment by tulinkry** [comment]
I'll just not it here, now something like `/search?group=abcd&q=foo` should work including all projects in group `abcd`.

It's not connected to UI yet.

File: opengrok-indexer/src/test/java/org/opengrok/indexer/web/PageConfigRequestedProjectsTest.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    /**
+     * Create a request with the specified path elements.
+     *
+     * @return a servlet request for the specified path
+     */
+    private static HttpServletRequest createRequest(
+            final String[] projects,
+            final String[] groups) {
+        final HttpServletRequest request = Mockito.mock(HttpServletRequest.class);
+        Mockito.when(request.getParameterValues("project")).thenReturn(projects);
```
Comment (vladak) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/web/PageConfigRequestedProjectsTest.java]: use the static final values here

Comment (tulinkry) [reply opengrok-indexer/src/test/java/org/opengrok/indexer/web/PageConfigRequestedProjectsTest.java]: done

**Pull request merged**